lib/requestability_resolver.js

```diff
@@ -1,21 +1,43 @@
 const DeliveryLocationsResolver = require('./delivery-locations-resolver')
+const { isItemNyplOwned } = require('./ownership_determination')
+const { isInRecap } = require('./util')
+const logger = require('./logger')
 class RequestabilityResolver {
   static fixItemRequestability (elasticSearchResponse) {
     elasticSearchResponse.hits.hits
       .map((hit) => {
         hit._source.items = hit._source.items.map((item) => {
           if (item.electronicLocator) return item
           let deliveryInfo
-          if (item.recapCustomerCode && item.recapCustomerCode[0]) {
+          const itemIsInRecap = isInRecap(item)
+          let physRequestableCriteria
+          if (itemIsInRecap && !item.recapCustomerCode) {
+            // recap items missing codes should default to true for phys and edd
+            // requestable.
+            physRequestableCriteria = 'Missing customer code'
+            deliveryInfo = { eddRequestable: true, deliveryLocation: [''] }
+          }
+          if (itemIsInRecap && item.recapCustomerCode && item.recapCustomerCode[0]) {
             deliveryInfo = DeliveryLocationsResolver.getRecapDeliveryInfo(item)
-          } else {
+            physRequestableCriteria = `${deliveryInfo.deliveryLocation &&
+              deliveryInfo.deliveryLocation.length || 0} delivery locations.`
+          }
+          if (!itemIsInRecap) {
             deliveryInfo = DeliveryLocationsResolver.getOnsiteDeliveryInfo(item)
+            physRequestableCriteria = `${deliveryInfo.deliveryLocation &&
+              deliveryInfo.deliveryLocation.length || 0} delivery locations.`
           }
-          const hasBarcode = (item.identifier || []).some((identifier) => /^(urn|bf):[bB]arcode:\d+/.test(identifier))
           item.eddRequestable = !!deliveryInfo.eddRequestable
           item.physRequestable = !!(deliveryInfo.deliveryLocation &&
-            deliveryInfo.deliveryLocation.length && hasBarcode)
+            deliveryInfo.deliveryLocation.length)
           item.specRequestable = !!item.aeonUrl
+          // items without barcodes should not be requestable
+          const hasBarcode = (item.identifier || []).some((identifier) => /^(urn|bf):[bB]arcode:\d+/.test(identifier))
```

Comment:
Looks like this `hasBarcode` calculation was borrowed from [here](https://github.com/NYPL/discovery-api/blob/fc800cfe3e922ca3cf660c745f8f2a57739182fc/lib/delivery-locations-resolver.js#L103-L104), where it's only run on NYPL (on-site) items, which as a rule only have numeric barcodes. Some partner barcodes contain leading letters. See cb10847813 (prod), which should be physRequestable but will fail the hasBarcode check until the regex accommodates non-numeric barcodes via something like:
```
  const hasBarcode = (item.identifier || []) 
    .some((identifier) => /^(urn|bf):[bB]arcode:\w+/.test(identifier))
```

Other prod examples:
 - cb7498199
 - (Most CUL items, maybe?)
 - hb990000045490203941